### PR TITLE
Add tests for every plateform + clippy

### PR DESCRIPTION
**Pull request by curquiza**

### CODE REVIEW

**Comment by curquiza** [comment]
Does the old MeiliSearch test on Windows?

**Comment by MarinPostma** [comment]
> Does the old MeiliSearch test on Windows?

I thought we did, but apparently not?

**Comment by curquiza** [comment]
We indeed don't: https://github.com/meilisearch/MeiliSearch/blob/master/.github/workflows/test.yml
https://github.com/meilisearch/MeiliSearch/search?q=windows

I've just released my tests does not pass, I shouldn't have requested you for a reviewer still it's not fixed!
I'll try to add test for Windows as well!

**Comment by curquiza** [comment]
bors try

**Comment by curquiza** [comment]
Cannot add tests on Windows since we get the "There is not enough space on the disk." errror
https://github.com/meilisearch/transplant/runs/2402295402

**Comment by curquiza** [comment]
bors try

**Comment by curquiza** [comment]
bors try

File: meilisearch-http/src/index_controller/update_actor/update_store.rs [hunk 1]
```diff
@@ -13,16 +13,16 @@ use uuid::Uuid;
 use crate::helpers::EnvSizer;
 use crate::index_controller::updates::*;
 
-type BEU64 = heed::zerocopy::U64<heed::byteorder::BE>;
+type Beu64 = heed::zerocopy::U64<heed::byteorder::BE>;
```
Comment (MarinPostma) [inline meilisearch-http/src/index_controller/update_actor/update_store.rs:16]: This name should remain like this. add `#[allow(non_camel_case)]` on top of it should fix it

**Review by MarinPostma — CHANGES_REQUESTED** [review]

File: meilisearch-http/src/index_controller/update_actor/update_store.rs (hunk 1 above)
Comment (curquiza) [reply meilisearch-http/src/index_controller/update_actor/update_store.rs:16]: Not pb, I change this. By curiosity, could you quickly explain why? 🙂

Comment (MarinPostma) [reply meilisearch-http/src/index_controller/update_actor/update_store.rs]: this is an acronym for Big Endian U64, so it makes more sense that is is all uppercased
This is also how we refer to this type in milli, and meilisearch :)

**Review by MarinPostma — APPROVED** [review]
lgtm

bors merge

**Pull request merged**